### PR TITLE
Add more unit tests for SelectionPredicate

### PR DESCRIPTION
**Pull request by njuptlzf**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
add ut for `MatcherIndex`
add ut for `MatchesObjectAttributes`
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by njuptlzf** [comment]
@yliaog 
Thanks for review, can the pr be merged? : )

File: staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go [hunk 1]
```diff
@@ -123,3 +128,55 @@ func TestSelectionPredicate(t *testing.T) {
...
+			indexFields:   []string{},
+			excepted:      nil,
+		},
+		"B": {
+			labelSelector: "name=foo",
+			fieldSelector: "uid=12345",
+			indexLabels:   []string{},
+			indexFields:   []string{"uid"},
+			excepted:      []MatchValue{{IndexName: "f:uid", Value: "12345"}},
+		},
```
Comment (yliaog) [inline staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go:152]: nit: better to have another test case with matching label, but no matching field

Comment (njuptlzf) [reply staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go:152]: > nit: better to have another test case with matching label, but no matching field

done

File: staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go [hunk 2]
```diff
@@ -123,3 +128,62 @@ func TestSelectionPredicate(t *testing.T) {
 		}
 	}
 }
+
+func TestSelectionPredicateMatcherIndex(t *testing.T) {
+	testCases := map[string]struct {
+		labelSelector, fieldSelector string
+		indexLabels                  []string
+		indexFields                  []string
+		excepted                     []MatchValue
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go:137]:
```suggestion
		expected                     []MatchValue
```

File: staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go [hunk 3]
```diff
@@ -123,3 +128,62 @@ func TestSelectionPredicate(t *testing.T) {
...
+			excepted:      []MatchValue{{IndexName: LabelIndex("name"), Value: "foo"}},
+		},
+		"Match field and label": {
+			labelSelector: "name=foo",
+			fieldSelector: "uid=12345",
+			indexLabels:   []string{"name"},
+			indexFields:   []string{"uid"},
+			excepted:      []MatchValue{{IndexName: FieldIndex("uid"), Value: "12345"}, {IndexName: LabelIndex("name"), Value: "foo"}},
+		},
+	}
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go:167]: add a test case for a field selector and label selector that specify negative matches on index keys and ensure MatcherIndex does not include them

```
		"Negative match field and label": {
			labelSelector: "name!=foo",
			fieldSelector: "uid!=12345",
			indexLabels:   []string{"name"},
			indexFields:   []string{"uid"},
			expected:      nil,
		},
```

Comment (njuptlzf) [reply staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go:188]: > add a test case for a field selector and label selector that specify negative matches on index keys and ensure MatcherIndex does not include them
> 
> ```
> 		"Negative match field and label": {
> 			labelSelector: "name!=foo",
> 			fieldSelector: "uid!=12345",
> 			indexLabels:   []string{"name"},
> 			indexFields:   []string{"uid"},
> 			expected:      nil,
> 		},
> ```

done

File: staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go (hunk 2 above)
Comment (njuptlzf) [reply staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go]: fixed typo, thanks.

**Comment by liggitt** [comment]
thanks!

/lgtm
/approve

**Pull request merged**